extensions.tf

```diff
@@ -52,3 +52,22 @@ module "vm-bootstrap" {
   common_tags = var.tags
 }
 
+#--------------------------------------------------------------
+# AADSSHLoginForLinux for Linux extension
+#--------------------------------------------------------------
+resource "azurerm_virtual_machine_extension" "entra" {
+  count                      = var.deploy_entra_extension && lower(var.vm_type) == "linux" ? 1 : 0
+  name                       = "AADSSHLoginForLinux"
+  virtual_machine_id         = azurerm_linux_virtual_machine.linvm[0].id
+  publisher                  = "Microsoft.Azure.ActiveDirectory"
+  type                       = "AADSSHLoginForLinux"
+  type_handler_version       = var.aad_type_handler_version
+  auto_upgrade_minor_version = true
+}
+
+resource "azurerm_role_assignment" "admin-user" {
+  for_each             = var.deploy_entra_extension && lower(var.vm_type) == "linux" ? { for key, value in var.rbac_config : key => value } : {}
+  scope                = each.value.scope
+  role_definition_name = each.value.role_definition_name
+  principal_id         = each.value.principal_id
+}
```

Comment:
Do most of our service principals have the permission to create role assignments? I _thought_ most of the `DTS Bootstrap *` Service Principals only have contributor over their respective sub.

Reply:
Although that might just be Heritage - in which case we can ignore this

Reply:
Ignore me - the existing SP's are setup with the correct roles to allow this.